app/services/katello/rhsm_fact_parser.rb

```diff
@@ -0,0 +1,151 @@
+module Katello
+  REDHAT_ATOMIC_HOST_DISTRO_NAME = "Red Hat Enterprise Linux Atomic Host".freeze
+  REDHAT_ATOMIC_HOST_OS = "RedHat_Enterprise_Linux_Atomic_Host".freeze
+
+  class RhsmFactParser < ::FactParser
+    def architecture
+      name = facts['lscpu.architecture'] || facts['uname.machine']
+      name = "x86_64" if name == "amd64"
+      name = "i386" if name == "i686"
+      Architecture.where(:name => name).first_or_create if name.present?
+    end
+
+    def model
+      if facts['virt::is_guest'] == "true"
```

Comment:
I know this is a direct import, but I'd expect `virtual` to be normalized to a real bool and reuse it here:
```suggestion
      if virtual
```